packages/core/src/metadata/MetadataDiscovery.ts

```diff
@@ -441,7 +441,8 @@ export class MetadataDiscovery {
 
       if (value instanceof Function) {
         const meta2 = this.metadata.get(prop.type);
-        prop[type] = value(meta2.properties).name;
+        
+        prop[type] = value(meta2.properties)?.name ?? null;
```

Comment:
we really need just that one char change i proposed, there is no need for explicit `null` fallback

```suggestion
        prop[type] = value(meta2.properties)?.name;
```